### PR TITLE
Add test cases for test_storages.py

### PR DESCRIPTION
**Pull request by himkt**
During the check of our documentation for exceptions (🔗  #2978), I found we can increase a coverage of `test_storages.py`. There are missing test cases that check if the exception documented in base class is actually raised.

This PR updates `test_storages` to add additional test cases for improving the coverage.
It adds tests for exception of the following methods:

- test_get_study_id_from_trial_id
- get_study_directions
- get_study_system_attrs
- get_trial_number_from_id
- set_trial_state
- check_trial_is_updatable

### CODE REVIEW

File: optuna/storages/_in_memory.py [hunk 1]
```diff
@@ -287,7 +287,7 @@ def get_best_trial(self, study_id: int) -> FrozenTrial:
             if best_trial_id is None:
                 raise ValueError("No trials are completed yet.")
             elif len(self._studies[study_id].directions) > 1:
-                raise ValueError(
```
Comment (himkt) [inline optuna/storages/_in_memory.py:290]: Although `BaseStorage` says `RuntimeError` is raised when `get_best_trial` is called for the multi-objective study, it actually raises `ValueError`. I fixed the mismatch even this PR only aims to improve the coverage.

https://github.com/optuna/optuna/blob/c9ea7abc081253e01c54b15d7019bbbeac4c97d6/optuna/storages/_base.py#L607-L608

Comment (himkt) [reply optuna/storages/_in_memory.py]: Update: this fix is extracted to https://github.com/optuna/optuna/issues/3491.
Reverted in [41f82cd](https://github.com/optuna/optuna/pull/3480/commits/41f82cd92d7ccffe1637b30c79ac59596311afe9).

**Comment by himkt** [comment]
Could you please give me reviews @HideakiImamura @nzw0301?

**Comment by nzw0301** [comment]
> Could you please give me reviews HideakiImamura @nzw0301?

Sure, the test file looks a little bit long; I'll take a little bit of time to check this PR.

File: tests/storages_tests/test_storages.py [hunk 2]
```diff
@@ -289,6 +294,10 @@ def check_set_and_get(key: str, value: Any) -> None:
         # Test overwriting value.
         check_set_and_get("dataset", "ImageNet")
 
+        # Non-existent study id.
+        with pytest.raises(KeyError):
+            storage.get_study_system_attrs(study_id + 1)
+
```
[the remark this answers is not in the document]
Comment (himkt) [reply tests/storages_tests/test_storages.py:301]: Thank you, I revised the PR in 8a2043c. Does it meat your intention?

[the remark this answers is not in the document]
Comment (nzw0301) [reply tests/storages_tests/test_storages.py:301]: Sorry, I meant using the code above because `non_existent_study_id` is clearer than `study_id + 1`.

**Review by nzw0301 — CHANGES_REQUESTED** [review]

**Comment by himkt** [comment]
> But this test checks whether or not to raise ValueError. Should we fix this?

Yes, I think. I extracted it into another issue (#3491) since I prefer to focus on adding test cases (🔗 https://github.com/optuna/optuna/pull/3480#discussion_r857065808). Thank you for creating the PR (#3541)!

File: tests/storages_tests/test_storages.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (himkt) [reply tests/storages_tests/test_storages.py:301]: Gotcha! I'll update, thank you!

[the remark this answers is not in the document]
Comment (himkt) [reply tests/storages_tests/test_storages.py]: [6fc72bc](https://github.com/optuna/optuna/pull/3480/commits/6fc72bc0310df38137c1f930bd4ca3b20e2b727d)

Assigning `non_existent_study_id` to `study_id + 1 `. I found other tests also use `study_id + 1` and defined variables if `study_id + 1` is used multiple times in each test.

File: tests/storages_tests/test_storages.py [hunk 3]
```diff
@@ -472,6 +488,14 @@ def test_set_trial_state(storage_mode: str) -> None:
             else:
                 assert storage.get_trial(trial_id).datetime_complete is None
 
+        # Non-existent study.
+        with pytest.raises(KeyError):
+            latest_trial_id = trial_ids[-1]
+            storage.set_trial_state(
+                latest_trial_id + 1,
```
Comment (nzw0301) [inline tests/storages_tests/test_storages.py:495]: Likewise

```suggestion
            non_existent_trial_id = trial_ids[-1] + 1
            storage.set_trial_state(
                non_existent_trial_id,
```

**Review by nzw0301 — CHANGES_REQUESTED** [review]

**Review by nzw0301 — APPROVED** [review]
Thanks!

File: tests/storages_tests/test_storages.py [hunk 4]
```diff
@@ -472,6 +488,14 @@ def test_set_trial_state(storage_mode: str) -> None:
             else:
                 assert storage.get_trial(trial_id).datetime_complete is None
 
+        # Non-existent study.
+        with pytest.raises(KeyError):
```
Comment (HideakiImamura) [inline tests/storages_tests/test_storages.py:492]: How about adding this type of test to the `test_set_trial_state_values_for_state`?

**Review by HideakiImamura — COMMENTED** [review]
Thanks for improving tests. I have a comment. PTAL.

File: tests/storages_tests/test_storages.py (hunk 4 above)
Comment (himkt) [reply tests/storages_tests/test_storages.py:492]: You got the point. I added the same test to `test_set_trial_state_values_for_state` in [ab89ad8](https://github.com/optuna/optuna/pull/3480/commits/ab89ad8d79d497759f9268b13067a62e367113f4).

**Review by HideakiImamura — APPROVED** [review]
Thanks for the update! LGTM!

**Pull request merged**